apkid/rules/dex/obfuscators.yara

```diff
@@ -109,3 +109,35 @@ rule bitwise_antiskid : obfuscator
     is_dex and
     any of them
 }
+
+
+rule arxan : obfuscator
+{
+  meta:
+    description = "Arxan"
+    info = "https://www.arxan.com/products/application-protection-mobile/"
+
+  strings:
+    $obf_package1 = "Lxxxxxx/"
+    $obf_package2 = "Ltttttt/"
+    $obf_package3 = "Loooooo/"
+    $obf_package4 = "Lpppppp/"
+    $obf_package5 = "Liiiiii/"
+    $obf_package6 = "Lffffff/"
+    $russian_char0 = "ШШШ"
+    $russian_char1 = "ййй"
+    $russian_char2 = "ЧЧЧ"
+    $russian_char3 = "ЯЯЯ"
+    $russian_char4 = "жжж"
+    $russian_char5 = "ЧЧ"
+    $russian_char6 = "лл"
+    $russian_char7 = "ии"
+    $russian_char8 = "ПП"
+    $russian_char9 = "УУ"
```

Comment:
I believe it would be best to have the non-utf-8 characters represented in their ascii codes. Otherwise they tend to get mangled.

Reply:
AFK this week, I will fix it next week.

Reply:
Hi @strazzere,

there are no ascii codes. Do you want me to encode the non-ascii characters in different encoding? Please give me more details.

Just a list with 8k characters (only < 128 are ascii):
```python
for i,c in enumerate(range(8192)):
	print ("{:5d}\t{}".format(i,chr(c)))
```

Normally, in this obfuscator you can expect characters like:
```
 1040	А
 1041	Б
 1042	В
 1043	Г
 1044	Д
 1045	Е
 1046	Ж
 1047	З
 1048	И
 1049	Й
 1050	К
 1051	Л
 1052	М
 1053	Н
 1054	О
 1055	П
 1056	Р
 1057	С
 1058	Т
 1059	У
 1060	Ф
 1061	Х
 1062	Ц
 1063	Ч
 1064	Ш
 1065	Щ
 1066	Ъ
 1067	Ы
 1068	Ь
 1069	Э
 1070	Ю
 1071	Я
 1072	а
 1073	б
 1074	в
 1075	г
 1076	д
 1077	е
 1078	ж
 1079	з
 1080	и
 1081	й
 1082	к
 1083	л
 1084	м
 1085	н
 1086	о
 1087	п
 1088	р
 1089	с
 1090	т
 1091	у
 1092	ф
 1093	х
 1094	ц
 1095	ч
 1096	ш
 1097	щ
 1098	ъ
 1099	ы
 1100	ь
 1101	э
 1102	ю
 1103	я
 1104	ѐ
 1105	ё
 1106	ђ
 1107	ѓ
 1108	є
 1109	ѕ
 1110	і
 1111	ї
 1112	ј
```

Reply:
I guess, I'd be curious if you have a dex file with these listed. Essentially I think it's bad form to keep them as "ascii" in a file and would prefer to have them hex encoded.

The main reason is because when I did a pull and just attempting to compile these rules, I lost the encoding due to just how my machine works without extra set up. I /could/ fix my machine to accept these characters, but fear we might hit an issue in the future if someone runs into it again.